### PR TITLE
Adding new flag 'new-namespace' for renaming apps(#165)

### LINKED ISSUE #165: kapp rename should support optional --new-namespace
this would help scenarios where state configmap was dropped into a wrong namespace.

### PR DESCRIPTION
**Pull request by renuy**

### CODE REVIEW

File: pkg/kapp/cmd/app/rename.go [hunk 1]
```diff
@@ -53,14 +55,27 @@ func (o *RenameOptions) Run() error {
 	if !exists {
 		return fmt.Errorf("%s", notExistsMsg)
 	}
+	newName := o.NewName
```
Comment (cppforlife) [inline pkg/kapp/cmd/app/rename.go:59]: nit: add newline before

File: pkg/kapp/cmd/app/rename.go [hunk 2]
```diff
@@ -53,14 +55,27 @@ func (o *RenameOptions) Run() error {
...
+	newNamespace := o.NewNamespace
+	if newName == "" && newNamespace == "" {
+		return fmt.Errorf("expected either new-name or/and new-namespace to be supplied")
```
Comment (cppforlife) [inline pkg/kapp/cmd/app/rename.go]: start error message with capital letter eg `Expected ...`. i would also add `--` in front of flag names.

File: pkg/kapp/cmd/app/rename.go [hunk 3]
```diff
@@ -53,14 +55,27 @@ func (o *RenameOptions) Run() error {
...
+		newName = app.Name()
+	}
+
+	if newNamespace == "" {
+		newNamespace = o.AppFlags.NamespaceFlags.Name
+	}
 
-	o.ui.PrintLinef("Renaming '%s' (namespace: %s) to '%s' (app changes will not be renamed)",
-		app.Name(), o.AppFlags.NamespaceFlags.Name, o.NewName)
+	o.ui.PrintLinef("Renaming '%s' (namespace: %s) to '%s (namespace: %s)' (app changes will not be renamed)",
```
Comment (cppforlife) [inline pkg/kapp/cmd/app/rename.go]: quotes are off in the added text from first usage (`'%s' (namespace: %s)`)

**Comment by renuy** [comment]
@cppforlife ,
2 more queries
1. If we rename the app, should the app history also reflect that. (there is todo in recorded_app.go (line 209)
2. should history be also moved to new namespace?

**Comment by cppforlife** [comment]
per our discussion, we ll leave app changes where they are.

**Pull request merged**